libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/ScheduledMailDialogRenderer.tsx

```diff
@@ -450,55 +458,66 @@ export class ScheduledMailDialogRendererUI extends React.PureComponent<
     };
 
     private onMessageChange = (value: string): void => {
-        this.emailBody = value;
+        this.setState({
+            emailBody: value,
+        });
     };
 
     private updateStartDateForRepeats(startDate: Date) {
-        const repeatType = this.repeatData.repeatType;
-        const repeatFrequency = this.repeatData.repeatFrequency;
-        const repeatExecuteOn = this.repeatData.repeatExecuteOn;
+        const { repeatType, repeatFrequency, repeatExecuteOn } = this.state.recurrency;
+
+        const newRecurrency = cloneDeep(this.state.recurrency);
 
         if (repeatType === REPEAT_TYPES.CUSTOM) {
             if (repeatFrequency === REPEAT_FREQUENCIES.MONTH) {
-                setMonthlyRepeat(this.repeatData, repeatExecuteOn, startDate);
+                setMonthlyRepeat(newRecurrency, repeatExecuteOn, startDate);
             } else if (repeatFrequency === REPEAT_FREQUENCIES.WEEK) {
-                setWeeklyRepeat(this.repeatData, startDate);
+                setWeeklyRepeat(newRecurrency, startDate);
             } else {
-                setDailyRepeat(this.repeatData);
+                setDailyRepeat(newRecurrency);
             }
         } else if (repeatType === REPEAT_TYPES.MONTHLY) {
-            setMonthlyRepeat(this.repeatData, REPEAT_EXECUTE_ON.DAY_OF_WEEK, startDate);
+            setMonthlyRepeat(newRecurrency, REPEAT_EXECUTE_ON.DAY_OF_WEEK, startDate);
         } else if (repeatType === REPEAT_TYPES.WEEKLY) {
-            setWeeklyRepeat(this.repeatData, startDate);
+            setWeeklyRepeat(newRecurrency, startDate);
         } else {
-            setDailyRepeat(this.repeatData);
+            setDailyRepeat(newRecurrency);
         }
+
+        this.setState({ recurrency: newRecurrency });
```

Comment:
Is this here because all the setters are mutating `newRecurrency`?

Reply:
yes